net/browserless_agent/src/browserless_agent.ts

```diff
@@ -0,0 +1,254 @@
+import { AgentFunction, AgentFunctionInfo, DefaultConfigData } from "graphai";
+
+interface BrowserlessInputs {
+  url: string;
+  operation?: "content" | "screenshot";
+  options?: Record<string, any>;
+  headers?: Record<string, string>;
+  cookies?: Array<{
+    name: string;
+    value: string;
+    domain?: string;
+  }>;
+}
+
+// Type for the buffer response from screenshot operations
+interface BufferResponse {
+  base64: string;
+  contentType: string;
+}
+
+interface BrowserlessParams {
+  debug?: boolean;
+  type?: "json" | "text" | "buffer";
+  throwError?: boolean;
+}
+
+type BrowserlessResult =
+  | object
+  | string
+  | BufferResponse
+  | {
+      onError?: {
+        message: string;
+        status?: number;
+        error: any;
+      };
+    };
+
+export const browserlessAgent: AgentFunction<BrowserlessParams, BrowserlessResult, BrowserlessInputs, DefaultConfigData> = async ({ namedInputs, params }) => {
+  const { url, operation, options, headers, cookies } = namedInputs;
+  const throwError = params?.throwError ?? false;
+
+  const browserlessEndpoint = "https://chrome.browserless.io";
+  const browserlessToken = process.env.BROWSERLESS_API_TOKEN;
+
+  // Check if API token is provided
+  if (!browserlessToken) {
+    const errorMessage = "Browserless API token is required. Please set the BROWSERLESS_API_TOKEN environment variable.";
+    throw new Error(errorMessage);
+  }
+
+  // Build the endpoint with token
+  const endpoint = `${browserlessEndpoint}?token=${browserlessToken}`;
+
+  // Select appropriate endpoint based on operation type
+  const operationType = operation || "content";
+  let apiPath: string;
+
+  switch (operationType) {
+  case "screenshot":
+    apiPath = "/screenshot";
+    break;
+  case "content":
+  default:
+    apiPath = "/content";
+    break;
+  }
+
+  const fullEndpoint = `${endpoint}${apiPath}`;
+
+  // Build request body
+  const requestBody: Record<string, any> = {
+    url,
+    ...options,
+  };
+
+  if (cookies && Array.isArray(cookies) && cookies.length > 0) {
+    requestBody.cookies = cookies;
+  }
+
+  // Return request information in debug mode
+  if (params?.debug) {
+    return {
+      url: fullEndpoint,
+      method: "POST",
+      headers: headers || { "Content-Type": "application/json" },
+      body: requestBody,
+    } as BrowserlessResult;
```

Comment:
Will it cause an error if you don't cast?
It's safer to avoid casting as much as possible for better type safety.